src/SourceWatcherFuncApp/DiagSourceWatcher.cs

```diff
@@ -0,0 +1,87 @@
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Linq;
+using SourceWatcherFuncApp.Services;
+using SourceWatcherFuncApp.Utilities;
+using SourceWatcherFuncApp.Entities;
+namespace Diag.SourceWatcher
+{
+    public class DiagSourceWatcher
+    {        
+        private IGithubService githubService;
+
+        private ITableStorageService storageService;
+
+        private IConfigurationRoot config;
+
+        private IBlobService blobService;
+
+        public DiagSourceWatcher(IConfigurationRoot configurationRoot, IGithubService GithubService, 
+                                 ITableStorageService StorageService, IBlobService BlobService)
+        {
+            githubService = GithubService;
+            config = configurationRoot;
+            storageService = StorageService;
+            blobService = BlobService;
+        }
+
+        [FunctionName("DiagSourceWatcher")]
+        public async Task Run([TimerTrigger("0 0 0 * * *")]TimerInfo timerInfo, ILogger log)
+        {
+            log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+
+            log.LogInformation($"Loading directories from github branch  {config["Github:Branch"]}");
+
+            var githubDirectories = await githubService.DownloadGithubDirectories(config["Github:Branch"]);
+          
+            foreach (var githubdir in githubDirectories)
+            {
+
+                if (!githubdir.Type.Equals("dir", StringComparison.OrdinalIgnoreCase)) continue;
+                
+                var contentList = await githubService.DownloadGithubDirectories(branchdownloadUrl: githubdir.Url);
+                
+                var assemblyFile = contentList.Where(githubFile => githubFile.Name.EndsWith("dll")).FirstOrDefault();
+                var scriptFile = contentList.Where(githubfile => githubfile.Name.EndsWith(".csx")).FirstOrDefault();
+                var configFile = contentList.Where(githubFile => githubFile.Name.Equals("package.json", StringComparison.CurrentCultureIgnoreCase)).FirstOrDefault();
+
+                if (assemblyFile != null && scriptFile != null && configFile != null)
+                {
+
+                    log.LogInformation($"Getting content for Assembly file : {assemblyFile.Path}");
+                    var assemblyData = await githubService.GetFileContentStream(assemblyFile.Download_url);
+
+                    log.LogInformation("Reading detector metadata");
+                    var configFileData = await githubService.GetFileContentByType<DetectorEntity>(configFile.Download_url);
+                    var scriptData = await githubService.GetFileContentByType<string>(scriptFile.Download_url);
```

Comment:
do you need to download csx file?

Reply:
Yes its needed in the `EntityInvoker` part of `EntityMetadata`

Reply:
Yes, but since we are not compiling the scripts, you can leave the script part empty and save 1/3rd of the download calls (for csx files) every day.

Reply:
I see. Sure I will make that change